core/state/access_events.go

```diff
@@ -151,6 +153,36 @@ func (ae *AccessEvents) AddTxDestination(addr common.Address, sendsValue bool) {
 	ae.touchAddressAndChargeGas(addr, zeroTreeIndex, utils.CodeHashLeafKey, false)
 }
 
+// SlotMultigas returns the amount of gas to be charged for a cold storage access.
+func (ae *AccessEvents) SlotMultigas(addr common.Address, slot common.Hash, isWrite bool) (*multigas.MultiGas, error) {
```

Comment:
I would suggest fixing the existing functions in `access_events.go` instead of adding a new so that it will be easier to track changes in the next update from the go-ethereum upstream